setup.py

```diff
@@ -23,6 +33,58 @@ def read_requirements(file_path):
 version = current_date if package_name == "torchao-nightly" else "0.1"
 
 
+def get_extensions():
+    debug_mode = os.getenv('DEBUG', '0') == '1'
+    if debug_mode:
+        print("Compiling in debug mode")
+
+    use_cuda = torch.cuda.is_available() and CUDA_HOME is not None
+    extension = CUDAExtension if use_cuda else CppExtension
+
+    extra_link_args = []
+    extra_compile_args = {
+        "cxx": [
+            "-O3" if not debug_mode else "-O0",
+            "-fdiagnostics-color=always",
+        ],
+        "nvcc": [
+            "-O3" if not debug_mode else "-O0",
+        ]
+    }
+    if debug_mode:
+        extra_compile_args["cxx"].append("-g")
+        extra_compile_args["nvcc"].append("-g")
+        extra_link_args.extend(["-O0", "-g"])
+
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    extensions_dir = os.path.join(this_dir, "torchao", "csrc")
+    sources = [
+        os.path.join(extensions_dir, p)
```

Comment:
more of a nit but should this be a `recursive=True` so we can have kernels grouped by folders under the `cuda/` folder